src/glk/gridmap.cpp

```diff
@@ -4,13 +4,13 @@
 
 namespace glk {
 
-GridMap::GridMap(double resolution, int width, int height, const unsigned char* values, int alpha, ColorMode mode) {
+GridMap::GridMap(double resolution, int width, int height, const unsigned char* values, int alpha, ColorMode mode) : mode(mode) {
   std::vector<unsigned char> rgba(width * height * 4);
```

Comment:
I think we can remove the initialization code from the constructor and just call `update_color()` to avoid duplication if `update_color()` creates a new texture rather than re-using the texture buffer.